app/controllers/api/cloud_volumes_controller.rb

```diff
@@ -2,12 +2,37 @@ module Api
   class CloudVolumesController < BaseController
     include Subcollections::Tags
 
+    def create_resource(_type, _id = nil, data = {})
+      ext_management_system = ExtManagementSystem.find(data['ems_id'])
+
+      klass = CloudVolume.class_by_ems(ext_management_system)
+
+      validate = klass.validate_create_volume(ext_management_system)
+      raise validate[:message] unless validate[:available]
```

Comment:
@agrare Is this specific to CloudVolume?  Asking because if we have this pattern for checking things, then should be apply that to the other PRs, like [this one](https://github.com/ManageIQ/manageiq-api/pull/962/files#diff-aa014dae2b42e9d76dcfe82a22847466b59bdb401f5c066446059d2bee5410f2R6)?

Reply:
@Fryguy this is the older "AvailabilityMixin" style versus SupportsFeature